framework/decode/custom_vulkan_ascii_consumer.h

```diff
@@ -80,16 +80,52 @@ inline std::string EnumPointerDecoderToString(PointerDecoder<VkEnumType>* pObj)
     return pDecodedObj ? ('"' + util::ToString(*pDecodedObj) + '"') : "null";
 }
 
-template <typename T>
-inline std::string DataPointerDecoderToString(const T& pObj)
+/// Convert pointers to unknown types coming through as uint64_t.
+/// ValueDecoder::DecodeAddress() is turning these pointers into uint64_ts
+/// in code generated by `base_decoder_body_generator.py`
+/// These hold:
+/// `const struct AHardwareBuffer*` from
+/// `vkGetAndroidHardwareBufferPropertiesANDROID`, 'const void *' from
+/// `vkGetMemoryHostPointerPropertiesEXT`, and from
+/// `vkCmdSetCheckpointNV`, `IDirectFB*` from
+/// `vkGetPhysicalDeviceDirectFBPresentationSupportEXT`, `struct _screen_window*` from
+/// `kGetPhysicalDeviceScreenPresentationSupportQNX`, `Display*` from
+/// `vkGetPhysicalDeviceXlibPresentationSupportKHR`, and from
+/// `vkAcquireXlibDisplayEXT`, and from
+/// `vkGetRandROutputDisplayEXT`, `struct wl_display*` from
+/// `vkGetPhysicalDeviceWaylandPresentationSupportKHR`, `xcb_connection_t*` from
+/// `vkGetPhysicalDeviceXcbPresentationSupportKHR`, Win32 memory `HANDLE` from
+/// `vkGetMemoryWin32HandlePropertiesKHR`.
+/// The uint64_t is treated as an address in each case.
+inline std::string DataPointerDecoderToString(const uint64_t buffer)
 {
-    // NOTE : Currently we just output the integer value of data void pointers, for
-    //  fields that use a PointerDecoder<> we output the address of the decoder
-    //  itself...this isn't terribly useful, but the capture time address wouldn't
-    //  be a terribly useful thing anyway.  There are structures that have data
-    //  fields that can be interpreted correctly...if we want to output interpreted
-    //  void pointer data we can implement custom handlers for those structures.
-    return "\"" + util::PtrToString(pObj) + "\"";
+    std::string str{ "\"" + util::PtrToString(buffer) + "\"" };
+    return str;
+}
+
+/// Convert pointers to arrays of void pointers to a string of the captured address.
+template <typename T, typename U>
+inline std::string DataPointerDecoderToString(PointerDecoder<T, U>* pDecoder)
+{
+    if (nullptr == pDecoder)
+    {
+        assert(0 == 1);
+        return GFXRECON_TOJSON_NULL;
+    }
+    std::string str{ "\"" + util::PtrToString(pDecoder->GetAddress()) + "\"" };
+    return str;
+}
+
+/// Convert void pointers into a string representation of the address in the capture file.
+inline std::string DataPointerDecoderToString(PointerDecoder<uint8_t>* pDecoder)
+{
+    if (nullptr == pDecoder)
```

Comment:
Would this be called for `pData` parameter to `vkGetPipelineCacheData`?  That can legally be `nullptr` so this probably should just print "0x0" or some such for that case.  I wonder if you could just omit this `if` altogether.

Reply:
In that case the null would be returned from the `PointerDecoder<uint8_t>::GetAddress()`. `pDecoder` here should never be null, This parameter should really be a reference but the code generation delivers it as a pointer so I handle the null case just to be defensive.

Reply:
Done